pubpub/pubpub
# Edit Existing Pub Edges

This PR adds the ability to edit existing Pub Connections.

### UI changes

Connections are edited using a new 'Edit' button in the top right corner of a connection card in the connections dashboard page:

<img width="534" alt="Screen Shot 2021-10-04 at 10 25 09 AM" src="https://user-images.githubusercontent.com/6402908/135868990-a1cf7aea-5c55-427f-8f31-43a312116f8a.png">

Clicking this button will swap the read-only view of the connection to an editable view which functions identically to the NewEdgeEditor, with a couple of small changes (e.g. the 'Add connection' button is renamed to 'Save connection'):

<img width="545" alt="Screen Shot 2021-10-04 at 10 27 37 AM" src="https://user-images.githubusercontent.com/6402908/135869372-c7158af6-d27f-40ac-83d5-f64a8a9d8656.png">

### Metadata changes

Crossref's documentation states that in order to update metadata (e.g. relationships) you must first deposit content with empty metadata, then follow-up with a deposit containing the updated metadata. This PR also implements [this behavior](https://github.com/pubpub/pubpub/pull/1563/files#diff-3a45b33c886df538ea597c6daab024b1d2b5d6e0cfee2dce334669e6ec93e179R153).

In addition, for Pubs that already have been deposited, users are notified that updating a Pub's connections will also require a re-deposit for the connections to update in Crossref.

## Test plan

- Add a connection to a Pub. The Pub should have at least one release and a DOI.
- Go to the Pub's connection settings. You should see a notification that modifying the Pub's connections will require a manual re-deposit.
- Edit the connection using the new 'Edit' button. The connection should be updated and persisted.
- Go to the DOI section in the Pub's settings. The deposit preview should reflect the changes made to the Pub, i.e., the relationships should be updated.
- Deposit the Pub. Log in to https://test.crossref.org. You should see two deposit records in the queue: one with no relationship metadata, and one with the updated relationships.

-- 
~~WIP, just putting this here for visibility/early feedback.~~

~~My approach so far has been to refactor the `NewEdgeEditor` into a presentational component, while hoisting the fetch/async logic higher up into the dashboard. I then use the `NewEdgeEditor` within the `DasboardEdgesListing` component to swap an outbound edge component with the editor when an "edit" button is pressed.~~

~~Things left to do:~~
- [x] ~~Solicit feedback on approach/design~~
- [x] ~~Inform user that the updated Pub and connected Pubs will need to be redeposited~~
- [x] ~~Update deposit pipeline to first deposit with empty connection metadata, then follow up with a deposit with the newly updated metadata (according to guidelines in the [Crossref docs](https://www.crossref.org/documentation/metadata-stewardship/maintaining-your-metadata/updating-your-metadata/))~~
